### ISSUE TITLE
Create stricter Schema type to only allow structs

### ISSUE THREAD

**Issue by devmatteini** [doubleloop-io/effect-mongodb#18]
It would be nice to have errors when you try to pass any Schema but a Struct like mongodb expects.

```typescript
Db.collection("any", Schema.Struct({ name: Schema.String })) // ok

Db.collection("any", Schema.String) // error (already working)

Db.collection("any", Schema.Array(Schema.String)) // errors (not working)
```

**Comment by codingismy11to7** [comment]
yeah, i went down this path for a long time in my library and eventually abandoned it. would love to have the compiler check this. i think i mostly got it working at some point, but don't remember what went wrong

**Comment by devmatteini** [comment]
We did some experiments in this branch https://github.com/doubleloop-io/effect-mongodb/tree/strict-document-type, but at the end there were some edge cases that we didn't know how to fix.

Especially trying to support both `Schema.Struct` and `Schema.Class`.

But at the end, we preferred to finish the first version of the library, and then maybe in the future try to work on this again because it would be very cool
